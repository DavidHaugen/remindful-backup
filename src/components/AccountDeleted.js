import React, { Component } from 'react'
import remindfulApiService from '../services/remindful-api-service';
import TokenService from '../services/token-service'
import GoalsContext from '../context/GoalsContext'

class AccountDeleted extends Component {
  static contextType = GoalsContext

  state = {
    deleted: false
  }

  componentWillUnmount(){
    this.context.resetUser()
  }

  deleteAccount = () => {
    remindfulApiService.deleteUser()
    .then(TokenService.clearAuthToken())
      .then(this.setState({deleted: true}))
  }
  
  render(){
    return (
      <div className="main">
        <div className="wrapper">
          <h4>{'Your account has been successfully deleted'}</h4>\
        </div>
      </div>
    )
  }
}

export default AccountDeleted;