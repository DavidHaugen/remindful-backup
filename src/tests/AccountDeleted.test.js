import React from 'react';
import {MemoryRouter} from 'react-router-dom'
import AccountDeleted from '../components/AccountDeleted'
import {mount} from 'enzyme';
import renderer from 'react-test-renderer'

describe('<AccountDeleted />', () => {
  it('Renders without crashing', () => {
      mount(<MemoryRouter><AccountDeleted/></MemoryRouter>);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><AccountDeleted/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
});