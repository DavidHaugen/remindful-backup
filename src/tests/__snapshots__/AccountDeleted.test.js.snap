// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountDeleted /> renders the UI as expected 1`] = `
<div
  className="main"
>
  <div
    className="wrapper"
  >
    <h4>
      Your account has been successfully deleted
    </h4>
  </div>
</div>
`;
